packages/data-stores/src/domain-suggestions/selectors.ts

```diff
@@ -15,7 +15,15 @@ type DomainSuggestionSelectorOptions = Partial< Exclude< DomainSuggestionQuery,
 
 const createSelectors = ( vendor: string ) => {
 	function getCategories( state: State ) {
-		return state.categories;
+		// Sort domain categories by tier, then by title.
+		return [
+			...state.categories
+				.filter( ( { tier } ) => tier !== null )
+				.sort( ( a, b ) => ( a > b ? 1 : -1)  ),
+			...state.categories
+				.filter( ( { tier } ) => tier === null )
+				.sort( ( a, b ) => a.title.localeCompare( b.title ) ),
+		];
```

Comment:
Ideally, this would happen in the resolver or the reducer where the data is put into state.

As it is here, we transform the data as we extract if from state. We're also creating a new array each time the selector is called, which is generally discouraged because this will cause strict (referential) equality checks (`getCategories() !== getCategories()`).

If this transformation happens before inserting the data into state, we only need to apply the transformation once and the value returned from the selector will be consistent, passing strict equality checks.